pkgs/by-name/si/sillytavern/package.nix

```diff
@@ -0,0 +1,117 @@
+{ lib
+, buildNpmPackage
+, fetchFromGitHub
+, makeWrapper
+, pkgs
+}:
+
+let
+wrapperScript = pkgs.writeShellScriptBin "sillytavern-wrapper" ''
+  _name="sillytavern"
+  _BUILDLIBPREFIX="$(dirname "$(readlink -f "$0")")/../lib/sillytavern"
+  _SHAREPREFIX="$(dirname "$(readlink -f "$0")")/../share/sillytavern"
+  _config_dir="$HOME/.config/$_name"
+  _yellow_color_code="\33[2K\r\033[1;33m%s\033[0m\n\33[2K\r\033[1;33m%s\033[0m\n"
+  export NODE_PATH="$_BUILDLIBPREFIX/node_modules"
+
+  fusecleanup()
+  {
+    echo "Cleaning up fusemount"
+    fusermount -qzu "$_config_dir/opt"
+  }
+
+trap fusecleanup 1 2 3 6 15
+
+  fusecleanup
+  mkdir -p "$_config_dir/opt" "$_config_dir/optdiff"
```

Comment:
I wonder why they use optdiff? Is it necessary for the mounting?

Reply:
Who is "they" in this instance? Did you retrieve the contents of this wrapper script from somewhere else? I ask, because it seems quite complex for a node application.

Reply:
Ah, I answered my own question: https://mpr.makedeb.org/pkgbase/sillytavern/git/tree/sillytavern-wrapper.sh

Reply:
This package is indeed hard to package that is why It taked so much time + inspiration from the PKGBUILD and the wrapper script. Mostly because I never used this + of the nixos unmutability of nix store

Reply:
It appears that what the wrapper script is doing is creating a unified filesystem at `$_config_dir/opt`, made up of a read-only directory (`$_SHAREPREFIX` or `/usr/share/sillytavern`) with a read-write directory (`$_config_dir/optdiff`) overlaid on top of it.

I believe this means that any changes are written to files in `$_config_dir/optdiff`, while the defaults from `/usr/share/sillytavern` can be used.

This seems very complicated, and I'm not sure if it's all necessary. But it does work, and I'd suggest we just get it in while it's in a working state, and iterate from there.